### PR TITLE
Add key expiration to tailscale_tailnet_key

### PR DESCRIPTION
**Pull request by dizzzan**
**What this PR does / why we need it**:
Adds key expiration option to the `tailscale_tailnet_key` resource

**Which issue this PR fixes** 
N/A

**Special notes for your reviewer**:
See https://github.com/tailscale/tailscale-client-go/pull/34 for associated client change

### CODE REVIEW

**Review by DentonGentry — APPROVED** [review]

**Comment by dizzzan** [comment]
@DentonGentry Haven't looked too closely at your CI setup, but does this depend on the published client pkg?

**Comment by DentonGentry** [comment]
I created a v1.7.0 release in tailscale-client-go. You should be able to update to that version of the module and try again:
```
go get tailscale/tailscale-client-go@v1.7.0
git add go.mod go.sum
git commit --amend -S
git push -f
```

**Comment by DentonGentry** [comment]
dependabot sent a PR to update this repo to 1.7. You'll need to rebase your PR to pull that change in:
```
git rebase main
git push -f
```

**Review by DentonGentry — APPROVED** [review]

**Comment by DentonGentry** [comment]
> Didn't realize the docs were generated - should I remove my manual edits?

Yes, please remove the docs changes. If we want to update the docstring we'd go back to the source document.

**Comment by davidsbond** [comment]
You'll need to install the [tfplugindocs](https://github.com/hashicorp/terraform-plugin-docs) tool and then run `go generate ./...` which will update the documentation based on the code changes.

File: examples/resources/tailscale_tailnet_key/resource.tf [hunk 1]
```diff
@@ -2,4 +2,5 @@ resource "tailscale_tailnet_key" "sample_key" {
   reusable      = true
   ephemeral     = false
   preauthorized = true
+  expiry = 3600
```
Comment (davidsbond) [inline examples/resources/tailscale_tailnet_key/resource.tf:5]: Please run `terraform fmt` on this example just so the indentation is correct.

**Comment by DentonGentry** [comment]
Great! This looks good, and the CI is happy.

**Comment by dizzzan** [comment]
@DentonGentry @davidsbond thank you both for the guidance, and for a great service! :)

**Pull request merged**